src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -168,13 +176,12 @@ private async Task HandleClusterDatabaseChanged(string databaseName, long index,
                                 _serverStore.NotificationCenter.Storage.DeleteStorageFor(databaseName);
                             }
                         }
-                        else
-                        {
-                            await HandleSpecificClusterDatabaseChanged(databaseName, index, type, changeType, context, rawRecord, changeState);
-                        }
                     }
 
-                    // if deleted, unload / deleted and then notify leader that we removed it
+                    foreach (var toUpdate in updateList)
```

Comment:
There are at least two `return` statements before this, so it will not always be hit, sounds like a problem?

Reply:
done